src/main/starlark/core/repositories/bzlmod_setup.bzl

```diff
@@ -0,0 +1,62 @@
+"""Definitions for bzlmod module extensions."""
+
+load(
+    "@bazel_tools//tools/build_defs/repo:http.bzl",
+    "http_archive",
+)
+load(
+    "//src/main/starlark/core/repositories:initialize.bzl",
+    _kotlin_repositories = "kotlin_repositories",
+    _kotlinc_version = "kotlinc_version",
+    _ksp_version = "ksp_version",
+)
+load(
+    "//src/main/starlark/core/repositories:versions.bzl",
+    _versions = "versions",
+)
+
+_version_tag = tag_class(
+    attrs = {
+        "version": attr.string(mandatory = True),
+        "sha256": attr.string(mandatory = True),
+    },
+)
+
+def _extra_repositories():
+    # TODO(bencodes) This really needs to be a development dependency
```

Comment:
Left a TODO here to move this out into a development dependency. I'll do another pass once this is merged to get these dependencies into better shape so that we can do a production-ready release of rules_kotlin.